modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache

```diff
@@ -398,7 +398,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
       {{#items.isModel}}
       {{#required}}
       // ensure the json data is an array
-      if (!jsonObj.get("{{{baseName}}}").isJsonArray()) {
+      if (!jsonObj.get("{{{baseName}}}").isJsonArray(){{#isNullable}} && !jsonObj.get("{{baseName}}").isJsonNull(){{/isNullable}}) {
```

Comment:
thanks for the fix and the test.

shall the new condition `{{#isNullable}} && !jsonObj.get("{{baseName}}").isJsonNull(){{/isNullable}}` be put in its own code block instead of adding to the JSON array type check as the error message right below only mentions expecting the field to be an array?

Reply:
> thanks for the fix and the test.
> 
> shall the new condition `{{#isNullable}} && !jsonObj.get("{{baseName}}").isJsonNull(){{/isNullable}}` be put in its own code block instead of adding to the JSON array type check as the error message right below only mentions expecting the field to be an array?

Hi, I don't quite understand. It doesn't seem to make sense to bring up the nullable judgment alone, because nullable is just an auxiliary judgment. If we put in its own code block, the nullable judgment is set outside the original judgment. From the code point of view, the logic is the same. If you write a judgment to nullable alone to throw an exception, the whole logic will seem even more strange.

```mustache
     {{^isNullable}}
      if (jsonObj.get("{{baseName}}").isJsonNull()) {
        // throw exception here 
      }
      {{/isNullable}}
      // we will get an exception in the next judgement if json element is null
      // ensure the json data is an array
      if (!jsonObj.get("{{{baseName}}}").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `{{{baseName}}}` to be an array in the JSON string but got `%s`", jsonObj.get("{{{baseName}}}").toString()));
      }
```

Or this is the change you want.
``` mustache
{{#isNullable}} 
if  (!jsonObj.get("{{baseName}}").isJsonNull()) {
  // ensure the json data is an array
        if (!jsonObj.get("{{{baseName}}}").isJsonArray()) {
          throw new IllegalArgumentException(String.format("Expected the field `{{{baseName}}}` to be an array in the JSON string but got `%s`", jsonObj.get("{{{baseName}}}").toString()));
        }
}
{{/isNullable}}

```

Reply:
can you please PM me via Slack when you've time?

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

Reply:
ok👌